src/popup-page/context/global.tsx

```diff
@@ -15,30 +15,38 @@ export class ActionError extends Error {
 
 interface State {
   beeApiUrl: string
+  postageBatchId: string | null
+  globalPostageBatchEnabled: boolean
 }
 
-interface Action<T = string> {
-  type: 'BEE_API_URL_SAVE' | 'BEE_API_URL_CHANGE'
-  newValue?: T
+interface Action {
+  type: 'BEE_API_URL_SAVE' | 'BEE_API_URL_CHANGE' | 'GLOBAL_POSTAGE_BATCH_SAVE' | 'GLOBAL_POSTAGE_BATCH_ENABLED_SAVE'
+  newValue?: unknown
```

Comment:
I'd suggest these to be separate types with a type composition so that you can have typed payloads. E.g.:
```ts
interface ActionBeeUrlSave {type: 'BEE_API_URL_SAVE', newValue: string}
interface ActionGlobalPostageBatchEnabledSave {type: 'GLOBAL_POSTAGE_BATCH_ENABLED_SAVE', newValue: boolean}

type Action = ActionBeeUrlSave & ActionGlobalPostageBatchEnabledSave
```

Reply:
good idea!